cpp/src/parquet/properties.h

```diff
@@ -757,6 +825,19 @@ class PARQUET_EXPORT WriterProperties {
     return false;
   }
 
+  bool bloom_filter_enabled() const {
+    // Note: We do not encourage enabling bloom filter for all columns. So
+    // default_column_properties_.bloom_filter_enabled is always false and
+    // cannot be altered by user. Thus we can safely skip checking it here.
+    return std::any_of(column_properties_.begin(), column_properties_.end(),
+                       [](auto& p) { return p.second.bloom_filter_enabled(); });
```

Comment:
```suggestion
                       [](auto& p) { return p.second.bloom_filter_enabled(); });
```
Please spell out the type for auto.

Reply:
Because `column_properties_` is an mapping, should I use a decltype or spell out the type?